src/test/java/org/folio/service/orders/OrderRolloverServiceTest.java

```diff
@@ -492,24 +491,24 @@ void shouldUpdateOrderLinesCostAndEncumbranceLinksWithExchangeRateAndPolCurrency
     Encumbrance encumbranceOneTime = new Encumbrance().withSourcePurchaseOrderId(orderId1).withSourcePoLineId(poLineId1)
       .withOrderType(Encumbrance.OrderType.ONE_TIME).withInitialAmountEncumbered(30.16d);
     Transaction transactionOneTime = new Transaction().withId(currEncumbrId1).withFromFundId(fundId1)
-      .withEncumbrance(encumbranceOneTime);
+      .withEncumbrance(encumbranceOneTime).withCurrency(systemCurrency);
     Encumbrance encumbranceOngoing2 = new Encumbrance().withSourcePurchaseOrderId(orderId2).withSourcePoLineId(poLineId2)
       .withOrderType(Encumbrance.OrderType.ONGOING).withInitialAmountEncumbered(30.16d);
     Transaction transactionOngoing2 = new Transaction().withId(currEncumbrId2).withFromFundId(fundId2)
-      .withEncumbrance(encumbranceOngoing2).withExpenseClassId(expClassId2);
+      .withEncumbrance(encumbranceOngoing2).withExpenseClassId(expClassId2).withCurrency(systemCurrency);
     Encumbrance encumbranceOngoing3 = new Encumbrance().withSourcePurchaseOrderId(orderId3).withSourcePoLineId(poLineId3)
       .withOrderType(Encumbrance.OrderType.ONGOING).withInitialAmountEncumbered(30.16d);
     Transaction transactionOngoing3 = new Transaction().withId(currEncumbrId3).withFromFundId(fundId3)
-      .withEncumbrance(encumbranceOngoing3).withExpenseClassId(expClassId3);
+      .withEncumbrance(encumbranceOngoing3).withExpenseClassId(expClassId3).withCurrency(systemCurrency);
 
     List<Transaction> encumbrances = List.of(transactionOneTime, transactionOngoing2, transactionOngoing3);
     doReturn(succeededFuture(encumbrances)).when(transactionService).getTransactions(anyString(), any());
 
     double exchangeEurToUsdRate = 0.82858d;
     doReturn(succeededFuture(systemCurrency)).when(configurationEntriesCache).getSystemCurrency(requestContext);
     ConversionQuery actQuery = ConversionQueryBuilder.of().setBaseCurrency(polCurrency).setTermCurrency(systemCurrency).set(RATE_KEY, exchangeEurToUsdRate).build();
-    ExchangeRateProvider exchangeRateProvider = Mockito.mock(ManualExchangeRateProvider.class);
-    ManualCurrencyConversion manualCurrencyConversion = new ManualCurrencyConversion(actQuery, exchangeRateProvider, ConversionContext.of());
+    ManualExchangeRateProvider exchangeRateProvider = Mockito.mock(ManualExchangeRateProvider.class);
+    ManualCurrencyConversion manualCurrencyConversion = new ManualCurrencyConversion(actQuery, exchangeRateProvider, ConversionContext.of(), ManualExchangeRateProvider.OperationMode.DIVIDE);
```

Comment:
is it possible to add OperationMode MULTIPLY cases too

Reply:
> is it possible to add OperationMode MULTIPLY cases too

Rollover should use OperationMode.DIVIDE in all cases when a manual POL Cost Exchange Rate is set. This is because we only set 1 manual exchange rate when there should actually be 2, for example:

With dynamic ECB/IMF exchange rates:
  Create an Encumbrance Transaction from a Purchase Order, AUD -> USD, ~0.66 exchange rate
  Get estimated total amount in Purchase Order currency, USD -> AUD, ~1.53 exchange rate

With manual exchange rate:
  Create an Encumbrance Transaction from a Purchase Order, AUD -> USD, ~0.66 exchange rate
  ...is not set anywhere, the only solution is to perform division by 0.66

Reply:
make sense, thanks